source/termination_criteria/steady_heat_flux.cc

```diff
@@ -0,0 +1,212 @@
+        Postprocess::internal::compute_heat_flux_through_boundary_faces (*this);
+
+      double local_boundary_fluxes = 0.0;
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->at_boundary(f))
+              {
+                const types::boundary_id boundary_indicator
+                  = cell->face(f)->boundary_id();
+                if (boundary_indicators.find(boundary_indicator) != boundary_indicators.end())
+                  local_boundary_fluxes += heat_flux_and_area[cell->active_cell_index()][f].first;
+              }
+
+      const double global_heat_flux_integral
+        = Utilities::MPI::sum (local_boundary_fluxes, this->get_mpi_communicator());
+
+      // Keep a list of times and heat fluxes at those times
+      time_heat_flux.emplace_back(this->get_time(), global_heat_flux_integral);
+
+      // If the length of the simulation time covered in the list is shorter than the
+      // specified parameter, we must continue the simulation
+      if ((time_heat_flux.size() <= 2)
+          ||
+          (time_heat_flux.back().first - time_heat_flux.front().first < necessary_time_in_steady_state))
+        return false;
+
+      // Remove old entries outside of current time window
+      internal::trim_time_heat_flux_list(necessary_time_in_steady_state,time_heat_flux);
+
+      // Scan through the list and calculate the min, mean and max heat flux
+      // We assume a linear change of heat flux between times
+      double flux_min, flux_max, flux_prev, time_prev, flux_sum=0, flux_mean, deviation_max;
+      flux_min = flux_max = flux_prev = time_heat_flux.front().second;
+      time_prev = time_heat_flux.front().first;
+      for (auto it=time_heat_flux.begin(); it!=time_heat_flux.end(); ++it)
+        {
+          flux_min = std::min(flux_min, (*it).second);
+          flux_max = std::max(flux_max, (*it).second);
+          flux_sum += (((*it).second + flux_prev)/2.0)*((*it).first-time_prev);
+          time_prev = (*it).first;
+          flux_prev = (*it).second;
+        }
+
+      flux_mean = flux_sum/(time_heat_flux.back().first-time_heat_flux.front().first);
+
+      // If the min and max are within the acceptable deviation of the mean,
+      // we are in steady state and return true, otherwise return false
+      deviation_max = std::max(flux_mean - flux_min, flux_max - flux_mean);
+
+      AssertThrow(std::abs(flux_mean) > std::numeric_limits<double>::min(),
+                  ExcMessage("The average heat flux is close to 0.0. The "
+                             "'steady state heat flux' plugin can not compute a "
+                             "relative deviation of heat flux in this case."));
+
+      if (deviation_max/flux_mean > allowed_relative_deviation)
+        return false;
```

Comment:
Just to be sure, `return true` will not be reached if the above condition is met (e.g., even though it is not within the conditional statement)?

Reply:
As soon as the `return false` is reached, the function will return the value false and no other code in the function will be executed. Was that the question?
If you want I can change this to 
```
      if (deviation_max/flux_mean > allowed_relative_deviation)
        return false;
      else
        return true;
```
I didn't think about what would be easier to read, I just copied it from an existing plugin.

Reply:
No fine as is, just wanted to make sure I understood. Thanks for the explanation!